packages/wallet/ui/src/components/ConnectionSettingsDialog.jsx

```diff
@@ -173,6 +174,17 @@ const ConnectionSettingsDialog = ({
   }
 
   const saveAndClose = () => {
+    if (
+      JSON.stringify(config) === JSON.stringify(connectionConfig) &&
+      config.smartConnectionMethod === SmartConnectionMethod.KEPLR
```

Comment:
check this first since it's cheaper (`cheapCheck && moreExpensiveCheck`)

Reply:
Just pulled this into a useMemo